src/json/formatter.cc

```diff
@@ -181,6 +181,10 @@ sstring formatter::to_json(unsigned long l) {
     return to_string(l);
 }
 
+future<> formatter::write(output_stream<char>& s, const jsonable& obj) {
+    return obj.write(s);
```

Comment:
That's some awesome change, but doesn't it create liveness issues described [here](https://github.com/scylladb/seastar/pull/2287#discussion_r1633145921)? So from now on the caller of formatter::weite(s, obj) needs to maintain the lifetime of `obj`, while prior to it it didn't have to? You fixed it in one place, but can there be others?

Reply:
> but doesn't it create liveness issues described https://github.com/scylladb/seastar/pull/2287#discussion_r1633145921

It goes back to the other comment thread. It's not clearly stated anywhere but in theory any user of `formatter::write` already had to guarantee this before because the same issue already exists on some of the other overloads like `write(output_stream<char>& s, const std::vector<Args...>&)`. 

My default assumption with async functions is that I have to make sure of lifetime issues unless the argument is by value. But not sure that's the correct default assumption.